internal/api/sms_provider/msg91.go

```diff
@@ -0,0 +1,122 @@
+package sms_provider
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/supabase/gotrue/internal/conf"
+)
+
+var err_codes = map[string]int{
+	"Missing mobile number": 101,
+	"Missing message":       102,
+	"Missing username":      104,
+	"Missing password":      105,
+	"Invalid username or password. 201 also appears in case the XML code triggered is incorrect.": 201,
+	"Invalid mobile number":                                         202,
+	"Invalid sender ID or DLT Entity Id Missing":                    203,
+	"Invalid authentication key":                                    207,
+	"IP is blacklisted":                                             208,
+	"Default route not found":                                       209,
+	"The route could not be determined. Please contact support":     210,
+	"DLT Template Id Missing":                                       211,
+	"The user does not have sufficient balance to send SMS":         301,
+	"Expired user account":                                          302,
+	"Banned user account":                                           303,
+	"This route is currently unavailable":                           306,
+	"The schedule time is incorrect":                                307,
+	"Campaign name cannot be more than 32 characters":               308,
+	"The selected group(s) does not belong to you":                  309,
+	"SMS is too long. The system paused this request automatically": 310,
+	"When the same SMS content is sent to the same number within 10 seconds. This is a security feature used to avoid multiple deliveries. The first SMS will be delivered and the second will be rejected. The balance will also be deducted only once.": 311,
+	"Flow ID Missing or Invalid Flow":                                 400,
+	"Flow Not Yet Approved":                                           401,
+	"Flow is disabled":                                                403,
+	"IP not whitelisted":                                              418,
+	"Internal error, please contact your account manager":             506,
+	"If your current route is disabled, kindly select another route":  602,
+	"This sender ID is blacklisted, please use a different sender ID": 603,
+	"Please enter at least one correct number to send an SMS":         604,
+	"The scheduled date cannot be more than three weeks":              606,
+	"Please enter the campaign name":                                  607,
+	"Scheduled SMS cannot be less than the current end time":          608,
+}
+
+const (
+	defaultMsg91ApiBase = "https://api.msg91.com/api/sendhttp.php?pluginsource=68"
```

Comment:
Is this supposed to have `?pluginsource=68`? Any links to their docs would be highly appreciated.

Reply:
as i have developed it with msg91 team hand in hand , they asked me to add this flag just for analytics usage, like they can know how many of their users are from supabase. If you say i can ask them to drop you official mail from the msg91 team confirming the same.

Reply:
Hey @MeetChaudhari we are not going to be able to merge this PR with this parameter. Please remove it!

Reply:
Yeah @MeetChaudhari, gotrue is meant to work independently from Supabase. Some of our users self-host gotrue and don't use Supabase but might want to use msg91 as an sms platform. I don't think it's a good idea to add a tracking param here and i don't think our users would appreciate that as well.

Reply:
I agree and i talked with msg91 team but this is the response form the msg91 team - 
"This is used to keep a track of requests coming in from Supabase. In case any error occurs, we  will be able to track the request quickly to provide a better experience.

This is majorly used to manage queues at our end and track any ongoing errors. 

It doesn't store or imply any customer details.
It just gives us report of total SMS failed(not user wise) with this Pluginsource so we can work on reducing number of failures and improve efficiency." 

what should i do?

Reply:
Unfortunately, you'll have to remove this one. If there are any issues with Msg91's service, we'll just ask the customer to reach out the Msg91 directly.

Reply:
ok i'll remove that.